posthog/api/cohort.py

```diff
@@ -51,15 +52,17 @@ def get_count(self, action: Cohort) -> Optional[int]:
         return None
 
 
-class CohortViewSet(viewsets.ModelViewSet):
+class CohortViewSet(StructuredViewSetMixin, viewsets.ModelViewSet):
+    legacy_team_compatibility = True  # to be moved to a separate Legacy*ViewSet Class
+
     queryset = Cohort.objects.all()
     serializer_class = CohortSerializer
     permission_classes = [IsAuthenticated, ProjectMembershipNecessaryPermissions]
 
     def get_queryset(self) -> QuerySet:
         queryset = super().get_queryset()
-        if self.action == "list":  # type: ignore
+        if self.action == "list":
             queryset = queryset.filter(deleted=False)
 
         queryset = queryset.annotate(count=Count("people"))
-        return queryset.filter(team=self.request.user.team).select_related("created_by").order_by("id")
+        return queryset.select_related("created_by").order_by("id")
```

Comment:
Q: Is this still correct? I am not seeing any code in StructuredViewSetMixin that is filtering the queryset to be limited to the team. Or is this what `get_parents_query_dict` is doing?

Reply:
Actually the `StructuredViewSetMixin` superclass `NestedViewSetMixin ` does the filtering in its `get_queryset` implementation (using `StructuredViewSetMixin`'s `get_parents_query_dict`).